pkg/adapter/client.go

```diff
@@ -6,47 +6,57 @@ import (
 
-	// I'm leaving this here for now as a document of the birdge format.  It can be removed later once the bridge
-	// function is more flushed out.
-	// bridge := `ssh -T root@192.168.122.1 "/usr/bin/varlink -A '/usr/bin/podman varlink \$VARLINK_ADDRESS' bridge"`
-	if len(r.cmd.RemoteHost) > 0 {
-		// The user has provided a remote host endpoint
+	// If the user defines an env variable for podman_varlink_bridge
+	// we use that as passed.
+	if bridge := os.Getenv("PODMAN_VARLINK_BRIDGE"); bridge != "" {
+		logrus.Debug("creating a varlink bridge based on env variable")
+		remoteEndpoint, err = newBridgeConnection(bridge, nil, r.cmd.LogLevel)
+		// if an environment variable for podman_varlink_address is defined,
+		// we used that as passed
+	} else if address := os.Getenv("PODMAN_VARLINK_ADDRESS"); address != "" {
+		logrus.Debug("creating a varlink address based on env variable")
+		remoteEndpoint, err = newSocketConnection(address)
+		//	if the user provides a remote host, we use it to configure a bridge connection
+	} else if len(r.cmd.RemoteHost) > 0 {
+		logrus.Debug("creating a varlink bridge based on user input")
 		if len(r.cmd.RemoteUserName) < 1 {
 			return nil, errors.New("you must provide a username when providing a remote host name")
 		}
-		remoteEndpoint.Type = BridgeConnection
-		remoteEndpoint.Connection = fmt.Sprintf(
-			`ssh -T %s@%s /usr/bin/varlink -A \'/usr/bin/podman --log-level=%s varlink \\\$VARLINK_ADDRESS\' bridge`,
-			r.cmd.RemoteUserName, r.cmd.RemoteHost, r.cmd.LogLevel)
-
-	} else if bridge := os.Getenv("PODMAN_VARLINK_BRIDGE"); bridge != "" {
-		remoteEndpoint.Type = BridgeConnection
-		remoteEndpoint.Connection = bridge
-	} else {
-		address := os.Getenv("PODMAN_VARLINK_ADDRESS")
-		if address == "" {
-			address = DefaultAddress
+		rc := remoteclientconfig.RemoteConnection{r.cmd.RemoteHost, r.cmd.RemoteUserName, false}
+		remoteEndpoint, err = newBridgeConnection("", &rc, r.cmd.LogLevel)
+		//  if the user has a config file with connections in it
+	} else if len(remoteConfigConnections.Connections) > 0 {
+		logrus.Debug("creating a varlink bridge based configuration file")
+		var rc *remoteclientconfig.RemoteConnection
+		if len(r.cmd.ConnectionName) > 0 {
+			rc, err = remoteConfigConnections.GetRemoteConnection(r.cmd.ConnectionName)
+		} else {
+			rc, err = remoteConfigConnections.GetDefault()
+		}
+		if err != nil {
+			return nil, err
 		}
-		remoteEndpoint.Type = DirectConnection
-		remoteEndpoint.Connection = address
+		remoteEndpoint, err = newBridgeConnection("", rc, r.cmd.LogLevel)
+		//	last resort is to make a socket connection with the default varlink address for root user
+	} else {
+		logrus.Debug("creating a varlink address based default root address")
+		remoteEndpoint, err = newSocketConnection(DefaultAddress)
 	}
 	return
 }
 
-// Connect provides a varlink connection
+// Connect provides a varlinit bbaude@REDHAT.COMnk connection
```

Comment:
Remove email address from string

Reply:
lol, that is a "hot" key from my keyboard.

Reply:
Kerberos strikes again